private-templates-service/client/src/components/Dashboard/PreviewModal/PreviewModal.tsx

```diff
@@ -66,6 +66,10 @@ class PreviewModal extends React.Component<Props, State> {
   };
 
   componentDidUpdate(prevProps: Props) {
+    if (!this.props.isFetching && (!this.props.template || !this.props.template.instances ||  this.props.template.instances.length === 0)) {
+      const history = this.props.history;
+      if (history) history.push("/");
```

Comment:
When does this happen, and should we add a `TODO` here for some sort of error popup?

Reply:
This should be expected behaviour rather than an error - if the last version of a template is deleted, then the user should be redirected to the dashboard